pandas/core/window/rolling.py

```diff
@@ -1326,21 +1293,29 @@ def count(self):
         # implementations shouldn't end up here
         assert not isinstance(self.window, BaseIndexer)
 
-        blocks, obj = self._create_blocks(self._selected_obj)
-        results = []
-        for b in blocks:
-            result = b.notna().astype(int)
+        _, obj = self._create_blocks(self._selected_obj)
```

Comment:
I was actually hoping that count could be defined in terms of `self._apply` so we can have less custom logic

```
def count(self):
    window_func = self._get_cython_func_type("roll_count")
    return self._apply(window_func, center=self.center, floor=0, name="count", **kwargs)
```

Might be not as straightforward (possibly) but just a heads up of where (hopefully) count should head.

Reply:
is this something you're planning on doing in the forseeable future?  i agree thatd be a nicer solution

Reply:
I'll experiment with this tonight and get a PR up if tests pass

Reply:
Looks like this version allows (inconsistently with other rolling function) non-numeric data like datetimes and strings because of the ahead of time coercing with `notna().astype(int)`. Taking longer than expected to tie out tests with using the `roll_count` cython version

Reply:
Not sure if anything can be shared, but it looks like `DataFrameGroupBy.count` is doing things kludgily blockwise and should also use an apply pattern